src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -105,15 +111,32 @@ public static void fillMetadata(Tangle tangle, TransactionViewModel transactionV
      * identifier, provided the {@link Transaction} exists in the database.
      *
      * @param tangle The tangle reference for the database
-     * @param hash The source that the {@link Hash} identifier will be created from, and the {@link Transaction} object
-     * will be fetched from the database from
+     * @param hash   The source that the {@link Hash} identifier will be created from, and the {@link Transaction}
+     *               object will be fetched from the database from
      * @return The {@link TransactionViewModel} with its Metadata filled in.
      * @throws Exception Thrown if the database fails to find the {@link Transaction} object
      */
     public static TransactionViewModel find(Tangle tangle, byte[] hash) throws Exception {
-        TransactionViewModel transactionViewModel = new TransactionViewModel(
-                (Transaction) tangle.find(Transaction.class, hash), HashFactory.TRANSACTION.create(hash));
+        Hash hashIdentifier = HashFactory.TRANSACTION.create(hash);
+        Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
+        TransactionViewModel transactionViewModel = null;
+
+        if (cache != null) {
+            transactionViewModel = cache.get(hashIdentifier);
+            if (transactionViewModel != null) {
+                fillMetadata(tangle, transactionViewModel);
```

Comment:
I am unsure whether we need to do this for something that is already cached

Reply:
Correction:
leave it and we will solve it on separate PR when we fix #1693

Reply:
ok